### ISSUE TITLE
George - Grupa A - Task 1-10

### ISSUE THREAD

**Issue by georgerusu017** [georgerusu017/churn_rate_predicition#7]

**Comment by georgerusu017** [comment]
# Imi da fail to upload. Nu stiu de ce, las textul


# George
# task 1
train_size = int(0.8 * len(df.sample(frac=1, random_state=45)))

train_df = df.iloc[:train_size]
test_df = df.iloc[train_size:]

train_df.to_csv("train.csv", index=False)
test_df.to_csv("test.csv", index=False)

# task 2 - ??? ce anume sa mentinem?

# task 3
print(f'Numar randuri train:{len(train_df)}\n',
      f'Numar randuri test:{len(test_df)}')

**Comment by georgerusu017** [comment]
import pandas as pd
from sklearn.model_selection import train_test_split


csv_path = "/Users/george/VSCProjects/Python3/Testing_ground/Proiect-comun/tranzactii.csv"


def validate_no_leakage(train_df, test_df):
    overlap = pd.merge(train_df, test_df, how="inner")
    if not overlap.empty:
        raise ValueError(
            f" Data leakage detected! {len(overlap)} rânduri comune între train si test.")
    else:
        print(" Nicio suprapunere de randuri între train si test.")

    print("\n Distributia clasei în train:")
    print(train_df['Class'].value_counts(normalize=True))
    print("\n Distributia clasei în test:")
    print(test_df['Class'].value_counts(normalize=True))


def sample_small_dataset(csv_path, frac=0.05, random_state=45):
    df = pd.read_csv(csv_path)
    small_df = df.sample(frac=frac, random_state=random_state)
    print(
        f"Esantion mic creat: {len(small_df)} randuri (din {len(df)} totale).")
    return small_df


def split_data(csv_path):
    df = pd.read_csv(csv_path)
    print(df.head())

    if len(df) < 50:
        raise ValueError(
            f"Dataset-ul are doar {len(df)} randuri. "
            "Prea putine date pentru split train/test!"
        )

    if "Class" in df.columns:
        train_df, test_df = train_test_split(
            df,
            test_size=0.2,
            random_state=45,
            stratify=df["Class"]
        )
    else:
        train_df, test_df = train_test_split(
            df,
            test_size=0.2,
            random_state=45
        )

    validate_no_leakage(train_df, test_df)

    train_df.to_csv("train.csv", index=False)
    test_df.to_csv("test.csv", index=False)

    print(f'Numar randuri train:{len(train_df)}\n'
          f'Numar randuri test:{len(test_df)}')


def show_median(test_csv, train_csv):
    df1 = pd.read_csv(test_csv)
    df2 = pd.read_csv(train_csv)

    print(f'Test \n {df1.mean(numeric_only=True)}')
    print(f'Train \n {df2.mean(numeric_only=True)}')


# split normal pe tot dataset-ul
split_data(csv_path)

train_csv = "/Users/george/VSCProjects/Python3/Testing_ground/Proiect-comun/train.csv"
test_csv = "/Users/george/VSCProjects/Python3/Testing_ground/Proiect-comun/test.csv"

show_median(test_csv, train_csv)
